Source/Blazorise/Components/Modal/Modal.razor.cs

```diff
@@ -313,26 +313,53 @@ protected virtual void HandleVisibilityStyles( bool visible )
                 } );
             }
 
-            DirtyClasses();
-            DirtyStyles();
+            await ModalDirtyStylingAnimation( visible );
+        }
+
+        /// <summary>
+        /// Handles the modal styling in a way that still triggers css animations.
+        /// </summary>
+        /// <param name="visible"></param>
+        /// <returns></returns>
+        private async Task ModalDirtyStylingAnimation( bool visible )
+        {
+            var hasFadeAnimation = !string.IsNullOrWhiteSpace( ClassProvider.ModalFade() );
+
+            if ( visible )
```

Comment:
Shouldn't both DirtyStyles and DirtyClasses be called? I don't we can assume that all providers would control animation with styles and classes in the same order.

Reply:
The order is changed according to visibility because of style block or none that would stop the animation.

Reply:
Srsly doubt it will affect. Otherwise we can just make the logic focus on the display style.